redfish-core/lib/hypervisor_system.hpp

```diff
@@ -591,9 +621,10 @@ inline void parseInterfaceData(nlohmann::json& jsonResponse,
                 }
             }
             ipv6Array.emplace_back(std::move(ipv6));
+            ipv6 = nlohmann::json::object();
```

Comment:
Similar comment like ipv4 case

Reply:
I observed that, there is Origin captured first and in the next loop, Origin is null, and in a later loop, the address is captured and in a later loop prefix length is captured.

I initially had the change you are suggesting, but the problem is that I want to push ipv6 details into the ipv6 array only when all the required fields are there in the json object. So I added a check for ipv6.size().
But the size is becoming 0 on every loop and it never reached the check in line 608.

Reply:
Question.

How/what would `ipv6Data` be provided - in case for one or more IP configurations?
Would each ip config field is separated each array element for the given IP, or each array element contain all fields?

You may add some debug statement to show the content of `ipv6Config` in each loop.

Reply:
So, there is a problem with the original code (as I mentioned in slack thread)

The outer for loop will loop through the getManaged Objects and the second for loop will loop through each interface implemented by that dbus object..
We will need IP interface only for the ipv4Data and IPv6Data to be filled. Instead, empty values are being pushed to the ipv4 and ipv6Data on every loop.

With my previous change in this PR, I can see just the following:
```

Aug 02 18:56:48 rain111bmc bmcweb[5264]: [DEBUG hypervisor_system.hpp:418] **** ipv4 size: 1, ipv6 size: 1
Aug 02 18:56:48 rain111bmc bmcweb[5264]: [DEBUG hypervisor_system.hpp:534] *** Inside for v4: isActive: true, origin: Static, netmask: 0.0.0.0, address: 0.0.0.0, ipv4Config.gateway: 0.0.0.0, ipv4.size(): 0
Aug 02 18:56:48 rain111bmc bmcweb[5264]: [DEBUG hypervisor_system.hpp:589] *** Inside for v6: isActive: false, origin: , address: ::, prefixLength: 128, size(): 0

```

But, with the original code, I can see the following (example ipv4):
```
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: , netmask: , address: , ipv4Config.gateway: , ipv4.size(): 0
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: , netmask: , address: , ipv4Config.gateway: , ipv4.size(): 0
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: , netmask: , address: , ipv4Config.gateway: , ipv4.size(): 0
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: , netmask: , address: , ipv4Config.gateway: , ipv4.size(): 0
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: Static, netmask: , address: , ipv4Config.gateway: , ipv4.size(): 0
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: false, origin: Static, netmask: , address: , ipv4Config.gateway: , ipv4.size(): 1
Aug 02 18:03:14 rain111bmc bmcweb[3525]: [DEBUG hypervisor_system.hpp:769] *** Inside for v4: isActive: true, origin: Static, netmask: 0.0.0.0, address: 0.0.0.0, ipv4Config.gateway: 0.0.0.0, ipv4.size(): 1
```

The above trace will be repeated for almost 33 times with empty value in most of the iterations.

This refactoring can be taken as a separate work/story I guess.